django_project/frontend/src/containers/MainPage/Metrics/TotalCountPerActivity.tsx

```diff
@@ -1,148 +1,162 @@
+        avColors[activityTypeName] = activityType.colour
+        activityTypeNames.push(activityTypeName)
       }
-    }else {
-      setBackgroundImageUrl(undefined)
+      setAvailableColors(avColors)
+      setActivityTypes(activityTypeNames)
     }
-  }, [propertyId,startYear,endYear,activityData, selectedSpecies]);
-
-  // Initialize variables
-  const labels: string[] = [];
-  const data: number[] = [];
-  const uniqueColors: string[] = [];
-  let year: number = endYear; // Set the year to the provided startYear
-
-  if (activityData && activityData.length > 0) {
-    // Iterate through activityData
-    activityData.forEach((speciesData: any) => {
-      const speciesActivities = speciesData.activities;
-
-      // Find the activity entry that matches the provided startYear
-      const matchingActivity = speciesActivities.find(
-          (activity: any) => activity.year === endYear? activity: null
-      );
-
-      if (matchingActivity) {
-        const activityType = matchingActivity.activity_type;
-        const total = matchingActivity.total;
-
-        // Check if the activityType is not in the labels list
-        if (!labels.includes(activityType)) {
-          let paddedLabel = activityType + '';
-          if (activityType.length > 25) {
-            // Trim the name to 22 characters and add '...' at the end
-            paddedLabel = activityType.substring(0, 22) + '...';
-          }
-
-          labels.push(paddedLabel.padEnd(50, ' ')); // Use the padded label
-          data.push(total);
-          uniqueColors.push(availableColors[labels.length - 1]);
-        }
-      }
-    });
-  }
+  }, [activityTypeList]);
 
-  // Create the chartData object
-  const chartData = processDataForChart(activityData);
+  // Create an array to hold datasets
+  const datasets = [];
 
-  // Define chart title based on conditions
-  let chartTitle = 'No data available for current filter selections';
+  for (const activityType of activityTypesObj) {
+    const count = activityData.filter((dataItem: any) => dataItem.activity_type === activityType.label)
+      .map((dataItem: any) => dataItem.total);
 
-  if (selectedSpecies && activityData && activityData.length > 0) {
-    chartTitle = `Total count per activity for ${selectedSpecies} year ${year}`;
-  }
+    if (count.every(d => d === 0)) continue;
+
+    const percentage = activityData.filter((dataItem: any) => dataItem.activity_type === activityType.label)
+      .map((dataItem: any) => parseFloat(((dataItem.total / totalPopulation) * 100).toFixed(2)));
 
-  if (!selectedSpecies){
-    chartTitle = "Please select a species for the chart to show available data";
+    // Create the dataset object
+    const dataset = {
+      label: activityType.label,
+      data: percentage,
+      counts: count,
+      backgroundColor: availableColors[activityType.label]
+    };
+
+    datasets.push(dataset);
   }
 
+  const data = {
+    labels: [''],
+    datasets: datasets,
+  };
+
+  const options = {
```

Comment:
@zamuzakki, this method could limit flexibility, as it assumes options are restricted to scales and datalabels with a predetermined structure. Consider instead developing a function that combines the BarChart component's default options with any new options.

Reply:
E.g. 

The original BarChart options :

```{ 'x' : 1, 'y': 2', 'z': {'a' : 2, 'b' : 3 }}```

A new component introduces additional options:

```{ 'z': {'c': 4, 'b': '5'}}```

Then the combined options will be :

```{ 'x' : 1, 'y': 2', 'z': {'a' : 2, 'b' : 5 , 'c':  4}}```

Reply:
@dimasciput I've updated it to use lodash merge
`const newOptions = _.merge(defaultOptions, options);`

![Screenshot from 2023-11-14 09-47-04](https://github.com/kartoza/sawps/assets/7352963/8342e3c5-a3b2-4524-9423-c64561decbbb)